src/config.rs

```diff
@@ -233,7 +233,9 @@ impl Config {
         };
 
         let magic = match (config.network, config.signet_magic) {
-            (Network::Signet, Some(magic)) => magic,
+            (Network::Signet, Some(magic)) => u32::from_str_radix(&magic, 16)
+                .expect("Invalid signet hex magic")
+                .swap_bytes(),
```

Comment:
Shouldn't this be `to_be` or `to_le`? I don't know the details of network magic and Core serialization but it does look suspicious.

Reply:
I'm no specialist either, but I made a little research about how people usually handle this kind of situations in Rust and this seemed the most straightforward way. I'll take a look again and try to make a better case for it (or switch to `to_le` if it's actually better).

Reply:
https://docs.rs/bitcoin/latest/bitcoin/network/constants/enum.Network.html#method.magic says it's encoded as little endian, so I think `u32::from_le` should be correct. Anyway, the API should be [improved on `rust-bitcoin` side](https://github.com/rust-bitcoin/rust-bitcoin/issues/1266).

Reply:
Sure, just note that the new version of rust-bitcoin wasn't released yet.

Reply:
Haha, yeah I know, just giving myself a little head start on this one